openedx/core/lib/teams_config.py

```diff
@@ -224,13 +226,16 @@ def cleaned_data(self):
         """
         JSON-friendly dictionary containing cleaned data from this TeamsConfig.
         """
-        return {
+        cleaned_data = {
             'id': self.teamset_id,
             'name': self.name,
             'description': self.description,
             'max_team_size': self.max_team_size,
             'type': self.teamset_type.value,
         }
+        if CONTENT_GROUPS_FOR_TEAMS.is_enabled(self._data.get('course_id')):
+            cleaned_data['dynamic_user_partition_id'] = self.dynamic_user_partition_id
+        return cleaned_data
```

Comment:
Since `self.dynamic_user_partition_id` already falls back to None, it should be safe to always include `'dynamic_user_partition_id'` in the returned dict, allowing you to remove this conditional.

Reply:
Thanks, that would be simpler. I added the conditional cause I thought it would confuse those examining the group configuration settings in Studio but if that's true, then it'd also apply when the flag is on. So I don't really find it relevant anymore. It's better to go with the simpler version, thanks again.